BLSQ/iaso
# POLIO-697 add error state to tabs

In the main polio form, when a field was in error because of bad data coming from the backend (ie: not input by the user), the user would be prevented to save, but would have no indication of why, or where the error lied

Related JIRA tickets : POLIO-697

## Self proofreading checklist


## Changes

- Made round dates optional
- Removed the complex validation on `Destruction`, `Shipment` and `FormA` in Vaccine tab
- For each tab, if there is a field in error, the tab text will appear in red. Exceptions: Scope and Preparedness, for which there was no error validation anyway.
- The above rule doesn't apply for sub tabs
- Changed error display rules: if the form is touched, all fields in error will switch to error state in order for user to see where the error lies when opening a "red" tab

## How to test

- Go to polio campaigns
- Open a campaign
- Add errors, see how the tabs react
- Do the same with creating a new campaign
- Do the same with a campaign created in admin with wrong data.

## Print screen / video

https://user-images.githubusercontent.com/38907762/218762721-569c3083-9383-4d5c-a508-9f99ce045846.mov


## Notes

I left the yup methods that were created for the validation of the Vaccine Management Tab so they can be reused if/when we move that tab out of the main polio modal
